portal-web/docroot/html/portal/layout.jsp

```diff
@@ -16,8 +16,16 @@
 
 <%@ include file="/html/portal/init.jsp" %>
 
+<%
+Boolean is404Page = (Boolean)request.getAttribute("IS_404_PAGE");
+
+if (is404Page == null) {
+	is404Page = Boolean.FALSE;
+}
+%>
+
 <c:choose>
-	<c:when test="<%= SessionErrors.contains(request, NoSuchLayoutException.class) %>">
+	<c:when test="<%= SessionErrors.contains(request, NoSuchLayoutException.class) && !is404Page %>">
```

Comment:
@4lejandrito @AliciaGarciaGarcia I've checked the other PR, and none of them makes my heart warm :-)

Wouldn't it be enough to update this condition to something like:

```
<c:when test="<% ... && !Objects.equals(request.getRequestURI(), PropsValues.LAYOUT_FRIENDLY_URL_PAGE_NOT_FOUND) %>">
```

There will be corner cases and all that, but from the outside, it seems unnecessary to set context stuff (paramenter, session error, etc.) because when the current URL is the not found one, we always want to display it, no matter what. Isn't that so?

Reply:
Hi @adolfopa we already tried that, it is the first thing I suggested to @AliciaGarciaGarcia. The problem is that at the point of rendering that JSP the request URI is no longer the one we want but `/portal/layout...`.

Reply:
In my on-live tests _request.getRequestURI()_ , on the point where it loads, will never be  _PropsValues.LAYOUT_FRIENDLY_URL_PAGE_NOT_FOUND_
will be _/c/portal/layout_